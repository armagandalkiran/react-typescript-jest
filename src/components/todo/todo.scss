.todoContainer {
  padding: 20px;
  display: grid;
  place-content: center;
}

.todoEntryWrapper {
}
