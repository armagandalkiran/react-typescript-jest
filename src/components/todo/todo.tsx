import { useEffect, useState } from "react";
import { ToDoItems } from "../../models/interfaces";
import { TodoProvider } from "../../providers/todoProvider";
import { TodoService } from "../../services/todoService";
import { TodoListItems } from "../todoListItems";
import { TextInput } from "../textInput";
import "./todo.scss";

const todoProvider = new TodoProvider();
const todoService = new TodoService(todoProvider);

export const ToDo = () => {
  const [text, setText] = useState("");
  const [tasks, setTasks] = useState<ToDoItems[]>([]);
  const [error, setError] = useState("");

  useEffect(() => {
    const getValues = async () => {
      const result = await todoService.getModifiedTodos();
      setTasks(result);
    };
    getValues();
  }, []);

  const displayError = () => {
    return error && <p className="todo__error--message">{error}</p>;
  };

  return (
    <div className="todo__container">
      <div className="todo__content">
        <TextInput
          text={text}
          setText={setText}
          setTasks={setTasks}
          setError={setError}
        />
        <div>{displayError()}</div>
        <ul className="todo__list--container">
          {tasks.map((data) => (
            <TodoListItems
              key={data._id}
              data={data}
              setTasks={setTasks}
              tasks={tasks}
            />
          ))}
        </ul>
      </div>
    </div>
  );
};
