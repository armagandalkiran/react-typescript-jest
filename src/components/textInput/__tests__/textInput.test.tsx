import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { ToDo } from "../../todo";

describe("TextInput tests", () => {
  test("entered input displayed in todolistitems component", () => {
    render(<ToDo />);
    const taskInput = screen.getByRole("textbox");
    const addButton = screen.getByRole("button", { name: "Ekle" });
    const nullItem = screen.queryByText(/testtt/i);
    expect(nullItem).not.toBeInTheDocument();
    userEvent.type(taskInput, "testtt");
    userEvent.click(addButton);
    const enteredItem = screen.getByText(/testtt/i);
    expect(enteredItem).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const tree = renderer.create(<ToDo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
