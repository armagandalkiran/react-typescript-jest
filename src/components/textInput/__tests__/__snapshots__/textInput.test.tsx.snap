// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests matches snapshot 1`] = `
<div
  className="todo__container"
>
  <div
    className="todo__content"
  >
    <div
      className="todo__entry--wrapper"
    >
      <input
        className="todo__input"
        data-testid="input"
        onChange={[Function]}
        placeholder="Enter your task.."
        type="text"
        value=""
      />
      <button
        className="todo__button"
        data-testid="button"
        onClick={[Function]}
      >
        Ekle
      </button>
    </div>
    <div />
    <ul
      className="todo__list--container"
    />
  </div>
</div>
`;
